stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/call/renderer/internal/SpotlightVideorenderer.kt

```diff
@@ -0,0 +1,276 @@
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment.Companion.CenterHorizontally
+import androidx.compose.ui.Alignment.Companion.CenterVertically
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.tooling.preview.Devices
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import io.getstream.video.android.compose.theme.VideoTheme
+import io.getstream.video.android.compose.ui.components.call.renderer.ParticipantVideo
+import io.getstream.video.android.compose.ui.components.call.renderer.SpotlightVideoRendererStyle
+import io.getstream.video.android.compose.ui.components.call.renderer.VideoRendererStyle
+import io.getstream.video.android.core.Call
+import io.getstream.video.android.core.ParticipantState
+import io.getstream.video.android.mock.StreamMockUtils
+import io.getstream.video.android.mock.mockCall
+import io.getstream.video.android.mock.mockParticipant
+import io.getstream.video.android.mock.mockParticipantList
+import me.saket.telephoto.zoomable.rememberZoomableState
+import me.saket.telephoto.zoomable.zoomable
+import java.lang.Integer.max
+
+@Composable
+internal fun SpotlightVideoRenderer(
+    modifier: Modifier = Modifier,
+    call: Call,
+    speaker: ParticipantState?,
+    participants: List<ParticipantState>,
+    orientation: Int = ORIENTATION_PORTRAIT,
+    isZoomable: Boolean = true,
+    style: VideoRendererStyle = SpotlightVideoRendererStyle(),
+    videoRenderer: @Composable (
+        modifier: Modifier,
+        call: Call,
+        participant: ParticipantState,
+        style: VideoRendererStyle,
+    ) -> Unit = { videoModifier, videoCall, videoParticipant, videoStyle ->
+        ParticipantVideo(
+            modifier = videoModifier,
+            call = videoCall,
+            participant = videoParticipant,
+            style = videoStyle,
+        )
+    },
+) {
+    if (participants.size == 1) {
+        // Just display the one participant
+        videoRenderer.invoke(
+            modifier.fillMaxSize().padding(VideoTheme.dimens.participantsGridPadding),
+            call,
+            participants[0],
+            style,
+        )
+        return
+    }
+
+    val derivedParticipants by remember(key1 = participants) {
+        derivedStateOf {
+            participants.filterNot {
+                it.sessionId == speaker?.sessionId
+            }
+        }
+    }
+    val listState =
+        lazyStateWithVisibilityNotification(call = call, original = rememberLazyListState())
+
+    BoxWithConstraints(modifier = modifier.fillMaxSize()) {
```

Comment:
Let's use `Box` instead of `BoxWithConstraints`. If we aren't using the constraints then we are paying a performance hit for no reason compared with using a plain Box.